File: Problems/0007. Reverse Integer/Solution.cpp
#include <bits/stdc++.h>

class Solution {
public:
    int reverse(int x) {
        int result = 0;

        while (x != 0) {
            // handle overflow/underflow
            if (result > INT_MAX / 10 || result < INT_MIN / 10) return 0;

            result = result * 10 + x % 10;
            x /= 10;
        }

        return result;
    }
};

File: Problems/0007. Reverse Integer/Solution.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let (mut result, mut n): (i32, i32) = (0, x);

        while n != 0 {
            // handle overflow/underflow
            if result > std::i32::MAX / 10  || result < std::i32::MIN / 10 {
                return 0;
            }

            result = result * 10 + n % 10;
            n /= 10;
        }

        return result;
    }
}